include/nuttx/drivers/himem_chardev.h

```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+ * include/nuttx/drivers/himem_chardev.h
```

Comment:
should we move the header file to arch/xtensa/include/esp32? since it is specific to esp32.

Reply:
Yes, please move the header file to arch/xtensa/include/esp32. Thank you very much for your suggestions!